src/Test.FunctionalTests.BinSkim.Rules/RuleTests.cs

```diff
@@ -131,12 +131,22 @@ public RuleTests(ITestOutputHelper output)
 
                 if (missingEntirely)
                 {
-                    sb.AppendLine("Expected '" + ruleName + "' " + expectedText + " but saw no result at all for file: " + Path.GetFileName(target));
+                    sb.AppendLine(
+                        string.Format("Expected '{0}' {1} but saw no result at all for file: {2]",
+                            ruleName,
+                            expectedText,
+                            Path.GetFileName(target)));
                 }
                 else
                 {
                     other.Remove(target);
-                    sb.AppendLine("Expected '" + ruleName + "' " + expectedText + " but check " + actualText + " for: " + Path.GetFileName(target));
+                    sb.AppendLine(
+                        string.Format(
+                            "Expected '{0}' {1} but check {2} for: {3}",
```

Comment:
>Expected '{0}' {1} but check {2} for: {3}", [](http://example.com/codeflow?start=29&length=43)

This format string looks odd to me, what is this error condition actually saying? Do you have an example we could review?

Reply:
"Expected 'DoNotModifyStackProtectionCookie' failure but check succeeded for: __security_cookies.exe"

I'm not entirely certain why we have this at all. As written, even if we have a regression in the code we shouldn't hit this condition.

Reply:
AFAIK, I think we should! Try moving a test file from the 'Failed' directory to the 'Pass' directory in your enlistment and we should hit this code path.

D:\src\binskim\src\Test.FunctionalTests.BinSkim.Rules\FunctionalTestsData\BA2012.DoNotModifyStackProtectionCookie

Reply:
By the way, if you'd really like to improve this code path, what I'd do is to leave an example as a source code comment.

// Generates message such as the following:
// Expected 'DoNotModifyStackProtectionCookie' failure but check succeeded for: __security_cookies.exe"

Please note that you appear to have added apostrophes where they aren't needed, for 'failure' on line 135 and line 145. Ideally, grab current state of these formatted strings from the debugger to make sure they are good (and once you confirm, put an example in the source code comment, then someone won't enquote something in apostrophes when they shouldn't (I led us here by making this suggestion, of course, without noting that not all dynamic args might be good to enquote).

Also, for extra points, please add the rule id to the output, as this is super helpful in locating the test binarys, the directories for which are prefixed with this info, e.g., :

// Expected 'BA20212:DoNotModifyStackProtectionCookie' failure but check succeeded for: __security_cookies.exe"